optimade/server/entry_collections/entry_collections.py

```diff
@@ -1,34 +1,87 @@
-from abc import abstractmethod
-from typing import Collection, Tuple, List
+from abc import abstractmethod, ABC
+from typing import Tuple, List, Union
+
+from fastapi import HTTPException
+from lark import Transformer
 
-from optimade.server.mappers import BaseResourceMapper
 from optimade.filterparser import LarkParser
 from optimade.models import EntryResource
-from optimade.server.query_params import EntryListingQueryParams
+from optimade.server.config import CONFIG
+from optimade.server.exceptions import BadRequest
+from optimade.server.mappers import BaseResourceMapper
+from optimade.server.query_params import EntryListingQueryParams, SingleEntryQueryParams
+
 
+class EntryCollection(ABC):
+    """ Backend agnostic base class for collections of EntryResources. """
 
-class EntryCollection(Collection):  # pylint: disable=inherit-non-class
     def __init__(
         self,
         collection,
         resource_cls: EntryResource,
         resource_mapper: BaseResourceMapper,
+        transformer: Transformer,
     ):
         self.collection = collection
         self.parser = LarkParser()
         self.resource_cls = resource_cls
         self.resource_mapper = resource_mapper
+        self.transformer = transformer
+
+    @abstractmethod
+    def __len__(self) -> int:
+        """ Returns the overall number of entries in the collection. """
+
+    @abstractmethod
+    def count(self, **kwargs) -> int:
+        """ Returns the number of entries matching the query specified
+        by the keyword arguments.
+
+        """
+
+    @abstractmethod
+    def find(
+        self, params: EntryListingQueryParams
+    ) -> Tuple[List[EntryResource], int, bool, set]:
+        """
+        Fetches results and indicates if more data is available.
+
+        Also gives the total number of data available in the absence of page_limit.
```

Comment:
I wonder if this link will work, if yes, great! If not, then maybe just wrap page_limit in code-quotations?
```suggestion
        Also gives the total number of data available in the absence of [`page_limit`][optimade.server.query_params.EntryListingQueryParams.page_limit].
```

Reply:
None of these links work for me when serving the docs locally, does it require extra config?

Reply:
Ah, yeah. So this will not work until #430 is merged.
Essentially, the link-anchor to this API reference needs to exist in the documentation for it to be valid, i.e., `page_limit` needs to be present in the documentation for this link to work. Which it will be with #430 :)